portfolio/src/main/java/com/google/sps/servlets/BusinessCardServlet.java

```diff
@@ -0,0 +1,173 @@
+import com.google.appengine.api.datastore.FetchOptions;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * This class will handle the storage/retrieval of information from the
+ * Business Card Drop on the portfolio homepage.
+ */
+@WebServlet(BusinessCardServlet.BIZ_CARD_URL)
+public class BusinessCardServlet extends HttpServlet {
+  // Endpoint to access BusinessCardServlet
+  public static final String BIZ_CARD_URL = "/biz-card";
+
+  // Store the BlobstoreService instance for the application. Same instance as other files.
+  private final BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+
+  // Store the Datastore Service instance. Will hold all BizCard entities.
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  /**
+   * Add a business card to the Datastore.
+   * DO NOT ACCESS DIRECTLY - should only be accessed by Blobstore.
+   * Point Client to the blobstore upload URL, which can be retrieved from the BlobstoreUrlServlet
+   *
+   * Request parameter must be named "bizCard".
+   *
+   * @param request HTTP request sent from client for POST request
+   * @param response HTTP response to be sent to client
+   * @throws IOException if an IO error occurs while the request is being processed by the servlet.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the blobKey of the image so it can be located in the blobstore.
+    // Possible Errors:
+    // NullPointerException - No images uploaded
+    // IllegalArgumentException - File was not an image
+    BlobKey bizCardBlobKey = null;
+    try {
+      bizCardBlobKey = getImageBlobKey(request, "bizCard");
+    } catch (FileNotFoundException e) {
+      response.sendError(400, e.getClass().getSimpleName() +
+          " No images were uploaded. Please select an image to upload");
+      return;
+    } catch (IllegalArgumentException e) {
+      response.sendError(400, e.getClass().getSimpleName() +
+          " File not supported. Please upload a BMP, GIF, ICO, JPEG, PNG or TIFF file.");
+      return;
+    }
+
+    // Add Business Card BlobKey to the datastore
+    // Possible Errors:
+    // IllegalArgumentException - verify that entity was created properly
+    // ConcurrentModificationException - should not happen. Verify entity is not being edited.
+    // DatastoreFailureException - check Google Cloud Platform for more info.
+    Entity businessCardEntity = new Entity("BizCard");
+    businessCardEntity.setProperty("bizCard", bizCardBlobKey.getKeyString());
+    try {
+      datastore.put(businessCardEntity);
+    } catch (Exception e) {
```

Comment:
gbuenoandrade@ - Thanks for letting me know the syntax - I didn't know you could do that

antmar@ - I just checked app engine, and it looks like 500 errors are thrown automatically! I'm going to keep the general Exception, since I have no control over datastore.put and it's failure wouldn't provide any useful information to send to the client (it's our problem, not the user's - their input would be fine). Changing to throw the error!